src/miner.cpp

```diff
@@ -177,7 +180,44 @@ bool BlockAssembler::isStillDependent(CTxMemPool::txiter iter)
     return false;
 }
 
+void BlockAssembler::onlyUnconfirmed(CTxMemPool::setEntries& testSet)
+{
+    for (CTxMemPool::setEntries::iterator iit = testSet.begin(); iit != testSet.end(); ) {
+        // Only test txs not already in the block
+        if (inBlock.count(*iit)) {
+            testSet.erase(iit++);
+        }
+        else {
+            iit++;
+        }
+    }
+}
+
+bool BlockAssembler::TestPackage(uint64_t packageSize, unsigned int packageSigOps)
+{
+    if (nBlockSize + packageSize >= nBlockMaxSize)
+        return false;
+    if (nBlockSigOps + packageSigOps >= MAX_BLOCK_SIGOPS)
+        return false;
+    return true;
+}
 
+// Block size and sigops have already been tested.  Check that all transactions
+// are final.
+bool BlockAssembler::TestPackageFinality(const CTxMemPool::setEntries& package)
+{
+    uint64_t potentialBlockSize = nBlockSize;
+    unsigned int potentialBlockSigOps = nBlockSigOps;
+    BOOST_FOREACH (const CTxMemPool::txiter it, package) {
+        if (!IsFinalTx(it->GetTx(), nHeight, nLockTimeCutoff))
+            return false;
+        potentialBlockSize += it->GetTxSize();
+        assert(potentialBlockSize < nBlockMaxSize);
+        potentialBlockSigOps += it->GetSigOpCount();
+        assert (potentialBlockSigOps < MAX_BLOCK_SIGOPS);
+    }
+    return true;
+}
```

Comment:
Nit: since block size and sigops have already been checked, it shouldn't be necessary to re-check and assert here.